tools/vcp_terminal.py

```diff
@@ -0,0 +1,648 @@
+
+This tool can be 'compiled' into a windows binary using the following commands
+    >>> from distutils.core import setup
+    >>> import py2exe
+    >>> setup(console=['vcp_terminal.py'])
+"""
+
+from __future__ import print_function
+from __future__ import division
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+import os
+import sys
+import time
+import usb.core as usb
+import threading
+import logging
+
+try:
+    import Queue as queue
+except ImportError:
+    import queue
+
+CDC_CMDS = {
+    "SEND_ENCAPSULATED_COMMAND": 0x00,
+    "GET_ENCAPSULATED_RESPONSE": 0x01,
+    "SET_COMM_FEATURE": 0x02,
+    "GET_COMM_FEATURE": 0x03,
+    "CLEAR_COMM_FEATURE": 0x04,
+    "SET_LINE_CODING": 0x20,
+    "GET_LINE_CODING": 0x21,
+    "SET_CONTROL_LINE_STATE": 0x22,
+    "SEND_BREAK": 0x23,   # wValue is break time
+}
+
+
+class _Getch(object):
+    """ Gets a single character from standard input.
+            Does not echo to the screen.
+    """
+
+    def __init__(self):
+        try:
+            self.impl = _GetchWindows()
+        except ImportError:
+            self.impl = _GetchUnix()
+
+    def __call__(self):
+        return self.impl()
+
+
+class _GetchUnix(object):
+    def __call__(self):
+        import sys
+        import tty
+        import termios
+        fd = sys.stdin.fileno()
+        old_settings = termios.tcgetattr(fd)
+        try:
+            tty.setraw(sys.stdin.fileno())
+            ch = sys.stdin.read(1)
+        finally:
+            termios.tcsetattr(fd, termios.TCSADRAIN, old_settings)
+        return ch
+
+
+class _GetchWindows(object):
+    def __init__(self):
+        import msvcrt
+
+    def __call__(self):
+        import msvcrt
+        return msvcrt.getch()
+
+
+class ComPort(object):
+
+    def __init__(self, usb_device, start=True):
+        self.device = usb_device
+        self._isFTDI = False
+        self._rxinterval = 0.005            # sec
+        self._rxqueue = queue.Queue()
+        self._rxthread = None
+        self._rxactive = False
+        self.baudrate = 9600
+        self.parity = 0
+        self.stopbits = 1
+        self.databits = 8
+
+        cfg = usb_device.get_active_configuration()
+
+        if self.device.idVendor == 0x0403:       # FTDI device
+            self._isFTDI = True
+            log.debug("Configuring as an FTDI device, no cmd itf")
+            cmd_itfs = None
+            data_itfs = list(usb.util.find_descriptor(
+                cfg, find_all=True,
+                custom_match=lambda e: (e.bInterfaceClass == 0xFF)))
+            data_itf = data_itfs[0]
+            itf_num = data_itf.bInterfaceNumber
+        else:
+            data_itfs = list(
+                usb.util.find_descriptor(
+                    cfg,
+                    find_all=True,
+                    custom_match=lambda e: (
+                        e.bInterfaceClass == 0xA)))
+            if len(data_itfs) == 0:
```

Comment:
IMO using len() makes the intent a little clearer, but I'll update this to conform to the project style.